test/testautovariables.cpp

```diff
@@ -2028,6 +2028,15 @@ class TestAutoVariables : public TestFixture {
               "    if (p != nullptr) {}\n"
               "}\n");
         ASSERT_EQUALS("", errout.str());
+
+        check("void f(const std::vector<S>& v) {\n"
```

Comment:
does this lack `template<typename T>` or is that intentional?

Reply:
I didn't manage to reduce the code to a compilable example, but turning it into a function template fixes that. Thanks.